### DISCUSSION TITLE
How can I set the cache directory

### DISCUSSION THREAD

**Discussion by CicadaDennis** [firedrakeproject/firedrake#2542] [Firedrake support] [answered]
I had installed firedrake on one of our HPC clusters and it seems that when regular users attempt to utilize certain functions, that the program attempts to write into a directory called .cache in the firedrake installation folder. This causes permission problems, because regular users are not allowed to write into the program installation folders. Is there a way to set an environment variable or a command line option so that firedrake will use a different directory? That way, it can be set it to a directory in the user's space and avoid this problem.
In this case, the error occured during the write command (leaving some stuff out here, since the issue is not about the computation):

W = FunctionSpace(mesh2,"Argyris",5)
u=Function(W)
solve(a == L, u)
File("helmholtz.pvd").write(u)

Thanks,
Cicada Dennis

**Comment by connorjward** [comment, marked as answer]
You need to set the environment variables `PYOP2_CACHE_DIR` and `FIREDRAKE_TSFC_KERNEL_CACHE_DIR`.

**Reply by CicadaDennis** [reply]
Thanks so much. I am able to set those and it seems to work.
Cicada
